spec/play_store/subscription_purchases/subscription_purchase_spec.rb

```diff
@@ -109,6 +109,34 @@
     end
   end
 
+  describe "canceled subscription by user" do
```

Comment:
Maybe you could add another test which confirms that `user_cancellation_time_millis` and `cancelled_at` return `nil`, in case the subscription wasn't canceled by the user. The reason why I think this could be important is because this `nil` could potentially be used for further conditionals, which would break if something "truthy" would be returned instead.